jujugui/static/gui/src/app/jujulib/test-bakery.js

```diff
@@ -250,5 +261,133 @@ describe('Bakery', () => {
       assert.equal(args[0], 'macaroon');
     });
   });
+
+  describe('interact handling', () => {
+    const getResponse = fail => {
+      let response = {
+        target: {
+          status: 200,
+          responseText: '',
+          response: '',
+        }
+      };
+      if (fail) {
+        response.target.status = 0;
+      }
+      return response;
+    };
+
+    it('accepts a visit page method', () => {
+      const params = {
+        visitPage: () => { return 'visits'; }
+      };
+      bakery = new jujulib.Bakery(client, storage, params);
+      assert.equal(bakery._visitPage(), 'visits');
+    });
+
+    it('opens the visit page', () => {
+      const windowStub = sinon.stub(window, 'open');
```

Comment:
Yeah, I can see that. However, I don't think rejiggering bakery is within scope on this branch. We can look at that sort of improvement once we've finished getting the things done that are blocking it from landing in `develop`.